core/src/main/java/com/exadel/aem/toolkit/core/optionprovider/services/impl/EnumOptionSourceResolver.java

```diff
@@ -0,0 +1,194 @@
+ */
+public class EnumOptionSourceResolver implements OptionSourceResolver {
+    private static final Logger LOG = LoggerFactory.getLogger(EnumOptionSourceResolver.class);
+    private static final String VALUE_ATTRIBUTE_KEY = "value";
+    private Class<? extends Enum<?>> clazz;
+
+    /**
+     * Default constructor
+     */
+    public EnumOptionSourceResolver() {
+    }
+
+    /**
+     * Constructor for testing
+     * @param clazz to be used instead of looking for it in {@link Bundle} array from {@link BundleContext}
+     */
+    public EnumOptionSourceResolver(Class<? extends Enum<?>> clazz) {
+        this.clazz = clazz;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Resource pathResolve(SlingHttpServletRequest request, PathParameters pathParameters) {
+        return resolve(request, pathParameters.getPath(), pathParameters);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Resource fallbackResolve(SlingHttpServletRequest request, PathParameters pathParameters) {
+        return resolve(request, pathParameters.getFallbackPath(), pathParameters);
+    }
+
+    /**
+     * Method, called by either {@link EnumOptionSourceResolver#resolve(SlingHttpServletRequest, PathParameters)} or
+     * {@link EnumOptionSourceResolver#fallbackResolve(SlingHttpServletRequest, PathParameters)} public methods.
+     * Method retrieves {@link Class} from {@link BundleContext} that should be {@link Enum} and then builds
+     * a {@link Resource} based on this Enum and {@link PathParameters}
+     * @param request {@link SlingHttpServletRequest} object
+     * @param path           should be a valid {@link String} class name.
+     * @param pathParameters represents key containing object to retrieve values from enum {@link Class}
+     * @return {@link Resource}
+     */
+    private Resource resolve(SlingHttpServletRequest request, String path, PathParameters pathParameters) {
+        if (clazz == null) {
+            setClass(path);
+        }
+
+        if (clazz == null) {
+            LOG.error("EnumOptionSourceResolver::resolve - value isn't a valid class name: {}", path);
+            return null;
+        }
+
+        if (!clazz.isEnum()) {
+            LOG.error("EnumOptionSourceResolver::resolve - class {} is not an Enum", clazz.getName());
+            return null;
+        }
+
+        if (clazz.getEnumConstants().length == 0) {
+            LOG.error("EnumOptionSourceResolver::resolve - Enum class {} doesn't have constants", clazz.getName());
+            return null;
+        }
+
+        List<Resource> children = new ArrayList<>();
+
+        for (Enum<?> enm : clazz.getEnumConstants()) {
+            children.add(new ValueMapResource(request.getResourceResolver(),
+                StringUtils.EMPTY,
+                StringUtils.EMPTY,
+                new ValueMapDecorator(buildEnumConstantParams(enm, pathParameters))));
+        }
+
+        clazz = null;
```

Comment:
We need to unset clazz field to make sure OptionProvider with more than one OptionSource, using Enum class, is supported.

Reply:
Do you mean a "tear down"? (a method to nullify the state of an OptionProvider between runs of tests)? 

Because in all the other situations the globally set "class" variable is not needed at all -- it will live strictly within the `resolve` method.

 If this is the teardown logic, then, it could make sense to move it to a separate method.
But then again, not absolutely needed, look at the screenshot 
![image](https://user-images.githubusercontent.com/48795460/201698494-f06a6917-f9e0-4f09-abb6-a5b7a7a8fec1.png)

When you want to pass some value that is not needed to be passed in "normal" situation, you can do this via the request attribute